policy_definitions/network_security/policy_definition_qby_VnetNamingConform.json

```diff
@@ -0,0 +1,161 @@
+{
+  "type": "Microsoft.Authorization/policyDefinitions",
+  "name": "QBY-Deny-Vnet-Subnet-NSG",
+  "properties": {
+    "displayName": "Vnet Subnet NSG Convention",
+    "description": "Test Vnet and Subnet For Naming Convention, and if the subnet has a rule that denys all inbound traffic. Checks if the subnet has the correct NSG attached to it.",
+    "policyType": "Custom",
+    "mode": "Indexed",
+    "metadata": {
+      "version": "1.0.0",
+      "category": "NetworkSecurity"
+    },
+    "parameters": {
+      "effect": {
+        "type": "String",
+        "defaultValue": "Deny",
+        "allowedValues": [
+          "Audit",
+          "Deny",
+          "Disabled"
+        ],
+        "metadata": {
+          "displayName": "Effect",
+          "description": "The effect determines what happens when the policy rule is evaluated to match"
+        }
+      }
+    },
+    "policyRule": {
+      "if": {
+        "allOf": [
+          {
+            "field": "type",
+            "equals": "Microsoft.Network/virtualNetworks"
+          },
+          {
+            "not": {
+              "allOf": [
+                {
+                  "field": "name",
+                  "like": "vnet-*"
+                },
+                {
+                  "value": "[length(split(field('name'), '-'))]",
+                  "greaterOrEquals": 7
+                },
+                {
+                  "count": {
+                    "field": "Microsoft.Network/virtualNetworks/subnets[*]",
+                    "where": {
+                      "not": {
+                        "allOf": [
+                          {
+                            "field": "Microsoft.Network/virtualNetworks/subnets[*].name",
+                            "like": "snet-*"
+                          },
+                          {
+                            "value": "[length(split(field('Microsoft.Network/virtualNetworks/subnets[*].name')[0], '-'))]",
+                            "greaterOrEquals": 7
+                          },
+                          {
+                            "value": "[equals(split(field('Microsoft.Network/virtualNetworks/subnets[*].networkSecurityGroup.id')[0],'/')[8], concat('nsg-', field('name')))]",
```

Comment:
Should we open a support case, why the `field('Microsoft.Network/virtualNetworks/subnets[*].networkSecurityGroup.id')[0]` is necessary?

Reply:
No,
turns out the examples use use a string() function that convertes the returned single element of the array as a string. Both ways are possible to get to the value